common/net/credentials/alts/conn/xchacha20poly1305ietf.go

```diff
@@ -0,0 +1,47 @@
+package conn
+
+import (
+	"fmt"
+
+	"github.com/GoKillers/libsodium-go/crypto/aead/xchacha20poly1305ietf"
+)
+
+type _xchacha20poly1305ietf struct {
+	key   [xchacha20poly1305ietf.KeyBytes]byte
+	nonce [xchacha20poly1305ietf.NonceBytes]byte
+}
+
+// NewxChaCha20Poly1305IETF creates an instance that uses chacha20poly1305 with rekeying
+// for ALTS record. The key argument should be 56 bytes, the first 32 bytes are used as
+// a secret key and the remainining 24 bytes are used as public nonce
+// Refer to: https://doc.libsodium.org/secret-key_cryptography/aead/chacha20-poly1305/xchacha20-poly1305_construction
+func NewxChaCha20Poly1305IETF(key []byte) (ALTSRecordCrypto, error) {
+	k := len(key)
+	if k != xchacha20poly1305ietf.KeyBytes+xchacha20poly1305ietf.NonceBytes {
+		return nil, KeySizeError(k)
+	}
+
+	c := &_xchacha20poly1305ietf{}
+	copy(c.key[:], key[:xchacha20poly1305ietf.KeyBytes])
+	copy(c.nonce[:], key[xchacha20poly1305ietf.KeyBytes:])
+	return c, nil
+}
+
+func (s *_xchacha20poly1305ietf) Encrypt(dst, plaintext []byte) ([]byte, error) {
+	ec := xchacha20poly1305ietf.Encrypt(plaintext, nil, &s.nonce, &s.key)
```

Comment:
you're nonce, key, additional data are same for every packages in same session - this not ensure replay attack, also could not verify integrity of data.
please check carefully aes128gcmRekey - this included hash verify + nounce as counter to ensure integrity + replay attack